src/nm-keyfile.c

```diff
@@ -138,32 +152,76 @@ netplan_render_yaml_from_nm_keyfile(GKeyFile* kf, const char* rootdir)
     }
     nd_type = type_from_str(type);
 
+    /* Use previously existing netdef IDs, if available, to override connections
+     * Else: generate a "NM-<UUID>" ID */
+    if (netdef_id)
+        nd_id = g_strdup(netdef_id);
+    else
+        nd_id = g_strconcat("NM-", uuid, NULL);
     nd = netplan_netdef_new(nd_id, nd_type, NETPLAN_BACKEND_NM);
+
+    /* Handle uuid & NM name/id */
+    nd->backend_settings.nm.uuid = g_strdup(uuid);
+    _kf_clear_key(kf, "connection", "uuid");
+    nd->backend_settings.nm.name = g_key_file_get_string(kf, "connection", "id", NULL);
+    if (nd->backend_settings.nm.name)
+        _kf_clear_key(kf, "connection", "id");
+
+    if (nd_type == NETPLAN_DEF_TYPE_OTHER)
+        goto only_passthrough; //do not try to handle any keys for connections types unknown to netplan
+
     /* remove supported values from passthrough, which have been handled */
     if (   nd_type == NETPLAN_DEF_TYPE_ETHERNET
         || nd_type == NETPLAN_DEF_TYPE_WIFI
+        || nd_type == NETPLAN_DEF_TYPE_MODEM
         || nd_type == NETPLAN_DEF_TYPE_BRIDGE
         || nd_type == NETPLAN_DEF_TYPE_BOND
         || nd_type == NETPLAN_DEF_TYPE_VLAN)
-        g_key_file_remove_key(kf, "connection", "type", NULL);
-
-    /* Handle uuid & NM name/id */
-    nd->backend_settings.nm.uuid = g_strdup(uuid);
-    g_key_file_remove_key(kf, "connection", "uuid", NULL);
-    nd->backend_settings.nm.name = g_key_file_get_string(kf, "connection", "id", NULL);
-    if (nd->backend_settings.nm.name)
-        g_key_file_remove_key(kf, "connection", "id", NULL);
+        _kf_clear_key(kf, "connection", "type");
 
     /* Handle match: Netplan usually defines a connection per interface, while
      * NM connection profiles are usually applied to any interface of matching
-     * type (like wifi/ethernet/...). Therefore, we match the interface on '*'
-     * if not specified. */
-    nd->match.original_name = g_key_file_get_string(kf, "connection", "interface-name", NULL);
-    if (nd->match.original_name)
-        g_key_file_remove_key(kf, "connection", "interface-name", NULL);
-    else
-        nd->match.original_name = g_strdup("*");
-    nd->has_match = TRUE;
+     * type (like wifi/ethernet/...). */
+    if (nd->type < NETPLAN_DEF_TYPE_VIRTUAL) {
+        nd->match.original_name = g_key_file_get_string(kf, "connection", "interface-name", NULL);
+        if (nd->match.original_name)
+            _kf_clear_key(kf, "connection", "interface-name");
+        /* Set match, even if it is empty, so the NM renderer will not force
+         * the netdef ID as interface-name */
+        nd->has_match = TRUE;
+    }
+
+    /* Modem parameters to detect GSM vs CDMA connections */
```

Comment:
hm, I don't think I understand this comment here. Below I only see handling of the GSM part - does this mean CDMA is to be handled via the passthrough? Is this what the comment implies?

Reply:
I've update the comment to explain the situation a bit better:
```
    /* Modem parameters
     * NM differentiates between GSM and CDMA connections, while netplan
     * combines them as "modems". We need to parse a basic set of parameters
     * to enable the generator (in nm.c) to detect GSM vs CDMA connections,
     * using its modem_is_gsm() util. */
```